.github/workflows/self-push-caller.yml

```diff
@@ -0,0 +1,30 @@
+name: Self-hosted runner (push-caller)
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - "src/**"
+      - "tests/**"
+      - ".github/**"
+      - "templates/**"
+      - "utils/**"
+
+jobs:
+  run_push_ci:
+    name: Run Push CI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout transformers
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+          ssh-key: "${{ secrets.COMMIT_KEY }}"
+
+      - name: Checkout to branch push-ci
+        run: |
+          git fetch origin
+          git checkout push-ci
+          git pull origin main
```

Comment:
Are you sure it updates the branch `push-ci` and not `main`? Okay for me as long as it's the case.